CONTRIBUTING.md

```diff
@@ -345,6 +345,25 @@ with clear instructions on how to install the required libraries.
 Furthermore, if your package requires CGO, it MUST be able to compile and operate in a no op mode
 or report a warning back to the collector with a clear error saying CGO is required to work.
 
+### Breaking changes
+
+Whenever possible, we adhere to semver as our minimum standards. Even before v1, we strive to not break compatibility
+without a good reason. Hence, when a change is known to cause a breaking change, it should be clearly marked in the
+changelog, possibly with a line instructing users how to move forward.
+
+We also strive to perform breaking changes in two stages: 
+
+- when we need to remove something, we mark a feature as deprecated in one version, and remove it in the next one
+- when renaming types, functions, or attributes, we create the new name first, deprecating the old one in one
+version, removing it in the next version
+- when creating new interfaces, provide helpers so that consumers can be forward compatible without much effort. One
+example can be found in the `configauth` package, where a [`ServerAuthenticator`](https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/config/configauth/serverauth.go#L28-L40)
+interface exists, we as well as a [helper](https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/config/configauth/default_serverauthenticator.go#L59-L71)
+that most implementations should use.
+
+When deprecating a feature affecting end-users, consider first deprecating the feature, then hiding it behind
```

Comment:
A feature flag should probably be added with the deprecation notice, but not yet disabling the feature.  Then the flag default can be set to disable the feature by default, and finally the flag and the feature removed.

Reply:
This is in line with what I had in mind. I changed the text to be clearer.

Reply:
Do you think the new version is satisfactory?

Reply:
I think it's still a little unclear that the feature gate should be added at the same time as the deprecation notice.  That would give users the option to toggle the gate to see how it functions with the feature removed before that becomes the default.  This somewhat extends the time available to detect issues with removing the feature and gives users an option beyond simply filing an issue saying that removing the deprecated behavior might impact them; they can actually test it and know.

Reply:
> give users the option to toggle the gate to see how it functions with the feature removed before that becomes the default

True, I haven't thought about that. How about this, each step being a version?

1. mark the feature as deprecated, add feature flag with the feature enabled by default
2. change the feature flag to disable it by default
3. remove the flag

Reply:
3. remove the flag and feature

but, yeah, that's in alignment with what I think will work best.

Reply:
Done. I changed the wording a bit, so, I'd appreciate another review of this part.